cassandra/policies.py

```diff
@@ -441,6 +458,85 @@ def on_add(self, host):
             RoundRobinPolicy.on_add(self, host)
 
 
+class HostFilterPolicy(LoadBalancingPolicy):
+    """
+    A :class:`.LoadBalancingPolicy` subclass that takes an instantiated
+    ``LoadBalancingPolicy`` as a child policy, and a single-argument predicate.
+    This policy defers to the child policy for hosts where ``predicate(host)``
+    is truthy. Hosts for which ``predicate(host)`` is falsey will be considered
+    :attr:`.IGNORED`, and will not be used in a query plan.
+    """
+
+    _predicate = None
+
+    def __init__(self, child_policy, predicate):
+        super(HostFilterPolicy, self).__init__()
+        self._child_policy = child_policy
+        self._predicate = predicate
```

Comment:
I might slightly prefer `except TypeError` when calling `self._predicate` and raising an error with a more useful error message. It's slightly more idiomatic, with the whole whole "better to ask forgiveness than permission" thing. Do you think one has technical merits over the other?